concordium-node/src/p2p/peers.rs

```diff
@@ -169,9 +175,143 @@ fn calculate_average_throughput(
     Ok((avg_bps_in, avg_bps_out))
 }
 
+/// The lmdb store name of the persisted peers.
+const PEERS_STORE_NAME: &str = "peers";
+
+/// A peer that is stored in the lmdb database.
+/// This is just a newtype over a [`SocketAddr`] so
+/// it is possible to implement the necessary serializing/deserializing traits
+/// used for putting and reading the peers in the database.
+#[derive(Debug, PartialEq, Eq)]
+pub struct StoredPeer(SocketAddr);
+
+impl From<SocketAddr> for StoredPeer {
+    fn from(addr: SocketAddr) -> Self { StoredPeer(addr) }
+}
+
+impl Serial for StoredPeer {
+    fn serial<W: Buffer + WriteBytesExt>(&self, target: &mut W) {
```

Comment:
Why are we implementing this? `serde` has instances for SocketAddr, so why are we not just using that instead of hacking our own serialization?